README
Copyright (C) 1998-2002 Eric Yeo. All rights reserved.
As of 2008 maintained by Jacob Nordfalk
This software has a LGPL license. Please see the file LICENSE for details.


DEVELOPMENT ENVIRONMENT
~~~~~~~~~~~~~~~~~~~~~~~
JMagick is developed and tested under Ubuntu Linux with OpenJDK 1.7.
It has not been tested on any other operating systems.

Please see top of Changelog.txt to see which which versions of ImageMagick
is currently supported.

BUILD WITH DOCKER
~~~~~~~~~~~~~~~~~~~~~~~
From source directory run:
	> docker build -t jmagick-builder .
	> docker run -v `pwd`:/src -v `pwd`/build-result:/build jmagick-builder

GIT REPOSITORY
~~~~~~~~~~~~~~
The JMagick source is now hosted on GitHub, although the mailing list still lives on SourceForge.
Source can be obtained by running

	> git clone git@github.com:techblue/jmagick.git


WEB SITES
~~~~~~~~~

Information on JMagick can be found at:

		https://github.com/techblue/jmagick
		https://sourceforge.net/projects/jmagick/
		https://sourceforge.net/p/jmagick/wiki/Home/

MAILING LIST
~~~~~~~~~~~~
The JMagick mailing list is jmagick-users@lists.sourceforge.net.
However, only list members are allowed to post to the mailing.
To join the JMagick mailing list, please point your web brower at
		https://lists.sourceforge.net/lists/listinfo/jmagick-users

Please make use of the mailing list for help and bug reports.



Jacob Nordfalk
1 July 2008

Andrew Kerr
20 June 2014


Changelog.txt
========== Compatibility statement for JMagick 6.5.7 ===========

The current verison of JMagick is compatible with ImageMagick version 6.4.1-2 and later.
Some new values have been added to support new functionality in IM 6.5.7-6, but if you don't use the new values then JMagick 6.5.7 should be compatible with any version if ImageMagick from 6.4.1-2 to 6.5.7-6.

----- issues with ImageMagick 6.4.1 and later ---------

In ImageMagick version 6.4.1-2 CompressionType's have been changed and therefore the class was changed to:
		public final static int UndefinedCompression = 0;
		public final static int NoCompression = 1;
		public final static int BZipCompression = 2;

		public final static int DXT1Compression = 3;  // Revision 10867 - 05/10/08 - version 6.4.1-2
		public final static int DXT3Compression = 4;  // IM Revision 10867
		public final static int DXT5Compression = 5;  // IM Revision 10867

		public final static int FaxCompression = 6;
		public final static int Group4Compression = 7;
		public final static int JPEGCompression = 8;
		public final static int JPEG2000Compression = 9;
		public final static int LosslessJPEGCompression = 10;
		public final static int LZWCompression = 11;
		public final static int RLECompression = 12;
		public final static int ZipCompression = 13;  // IM Revision 10867

This makes JMagick incompatible with ImageMagick version 6.4.1 and before, if any of the values greater than 2 is used or stored in your code.


----- issues with ImageMagick before version 6.3.9 ---------

There are only some very minor issues, namely that the classes ExceptionType
and GeometryFlags have been changed.
This *could* influence the return value of Magick.parseImageGeometry(),
but existing values are compatible, so code should not break.
IM Before version 6.2.9 does not support the DivideCompositeOp CompositeOperator.


========== Changelog ===========

2023-11-23
    * Added CI workflow support, configure.ac improved, see https://github.com/techblue/jmagick/commits/master for recent changes

2023-06-09
    * Added build support for Java 11, 17

2014-08-02 Christian Ferbar
	* Add Kerning, InterwordSpacing, InterlineSpacing, getTypeMetrics, FormatMagickCaption

2014-06-16 Andrew Kerr - JMagick 6.7.7
	* Enum interfaces changed to reflect ImageMagick 6.7.7 .h files.
	  ImageMagick 6.7.7 is the currently shipping version for Ubuntu 14.04 LTS.
	* Fix memory leak in MagickImage.imageToBlob(). (https://github.com/techblue/jmagick/issues/3)

2012-11-20 Andrew Kerr - JMagick 6.6.9
	* Enum interfaces changed to reflect ImageMagick 6.6.9 .h files.  The FilterType.java
	  file is no longer compatibile with older versions, due to changes to the corresponding
	  ImageMagick enum.  ImageMagick 6.6.9 is the currently shipping version for Ubuntu 12.04.1 LTS.

2012-11-14 Andrew Kerr - JMagick 6.6.2
	* Enum interfaces changed to reflect ImageMagick 6.6.2 .h files.  The QuantumTypes.java
	  file is no longer compatibile with older versions.
	* Added MagickImage.setQuality().  MagickInfo.setQuality() is ignored in newer ImageMagicks.
	  See http://sourceforge.net/mailarchive/forum.php?thread_name=alpine.GSO.2.01.1201091112090.20764%40freddy.simplesystems.org&forum_name=jmagick-users
	  for discussion.

2009-11-10 Jacob Nordfalk - JMagick 6.5.7
	* Memory leak bug fixed. Affected code is Java_magick_DrawInfo_setGeometry, Java_magick_DrawInfo_setFont, Java_magick_ImageInfo_setServerName, Java_magick_ImageInfo_setFont, Java_magick_ImageInfo_setSize, Java_magick_ImageInfo_setTile, Java_magick_ImageInfo_setDensity, Java_magick_ImageInfo_setPage, Java_magick_ImageInfo_setTexture, Java_magick_ImageInfo_setView, Java_magick_MontageInfo_setGeometry, Java_magick_MontageInfo_setTile, Java_magick_MontageInfo_setTitle, Java_magick_MontageInfo_setFrame, Java_magick_MontageInfo_setTexture, Java_magick_MontageInfo_setFont,


2009-11-09 Jacob Nordfalk
	* Added new entries to CompositeOperator CompressionType ExceptionType QuantumTypes InterlaceType NoiseType
	  which has been added from ImageMagick-6.5.7-6 to 6.4.1-2.
	  all entries are backwards compatible, i.e. no old values have changed


2008-12-??
	* Added setAntialias() and getAntialias()   (contrib from ???? - check
	mailing list)

2008-08-05
	* Added setImageOption()  (contrib from Scott Knight at Specialized Business Software)

2008-07-01  Jacob Nordfalk - JMagick 6.4.0
	* look for include files in both ${MPATH}/include/magick/api.h (old IM way) and
		${MPATH}/include/ImageMagick/magick/api.h  (IM 6.3.8.something and later)
	* Test run on Q16 (16 bits per pixed)

2008-05-22  Jacob Nordfalk
	* Small updates of javadoc

2008-??-??  Jacob Nordfalk
	* Added a test suite

2007-02-27  Jacob Nordfalk
	* Added methods setImageProfile and getImageProfile
	* Deprecated old and only partly working profile methods
	* Introduced package magicktest.BETA for test cases not yet finished

2007-02-25  Andy Lewis
	* Updated ExceptionType and GeometryFlags to reflect IM changes in 6.3.9
	* This makes JM incompatible ImageMagick versions before 6.3.9

2007-02-26  Jacob Nordfalk
	* Renamed test classes and polished the tests
	* Updated Win32 Makefile.all for Visual Studio 8.

2007-02-25  Andy Lewis
	* Updated CompositeOperator to reflect IM changes made to composite.h since 6.2.9
    (added public static final int DivideCompositeOp = 55;)
	* Added methods blurImageChannel() and trimImage()

2007-02-22  Jacob Nordfalk
	* First version on SourceForge.net
	* Renamed package test to magicktest and moved utility class MagickWindow to magick.util
	* configure.in now claims version is 6.3.8

2008-02-19  Eric O'Laughlen
	* A few tweaks to the output lib names to support version numbers and symlinks.
	For example, it outputs libJMagick-6.2.6.dylib and a link called libJMagick.dylib (.so's on Linux).
	An option to the configure.in offers an arg --with-shared-lib-ext=.dylib.

2007-11-26  Jacob Nordfalk
	* The ImageMagick engine wasnt initialized from JMagick!

src/magick/FilterType.java
package magick;

public interface FilterType {
    /*
        Important! Constant values should correspond to:
        http://git.imagemagick.org/repos/ImageMagick/blob/master/MagickCore/resample.h
    */
    public static final int UndefinedFilter = 0;
    public static final int PointFilter = 1;
    public static final int BoxFilter = 2;
    public static final int TriangleFilter = 3;
    public static final int HermiteFilter = 4;
    public static final int HannFilter = 5;
    public static final int HammingFilter = 6;
    public static final int BlackmanFilter = 7;
    public static final int GaussianFilter = 8;
    public static final int QuadraticFilter = 9;
    public static final int CubicFilter = 10;
    public static final int CatromFilter = 11;
    public static final int MitchellFilter = 12;
    public static final int JincFilter = 13;
    public static final int SincFilter = 14;
    public static final int SincFastFilter = 15;
    public static final int KaiserFilter = 16;
    public static final int WelchFilter = 17;
    public static final int ParzenFilter = 18;
    public static final int BohmanFilter = 19;
    public static final int BartlettFilter = 20;
    public static final int LagrangeFilter = 21;
    public static final int LanczosFilter = 22;
    public static final int LanczosSharpFilter = 23;
    public static final int Lanczos2Filter = 24;
    public static final int Lanczos2SharpFilter = 25;
    public static final int RobidouxFilter = 26;
    public static final int RobidouxSharpFilter = 27;
    public static final int CosineFilter = 28;
    public static final int SplineFilter = 29;
    public static final int LanczosRadiusFilter = 30;
    public static final int SentinelFilter = 31;  /* a count of all the filters, not a real filter */
}


